src/main/java/com/adyen/Client.java

```diff
@@ -63,6 +63,33 @@ public Client(String username, String password, Environment environment, String
         this.config.setApplicationName(applicationName);
     }
 
+    public Client(String username, String password, Environment environment, String applicationName, int connectionTimeoutMillis) {
+
+        this.config = new Config();
+        this.config.setUsername(username);
+        this.config.setPassword(password);
+        this.setEnvironment(environment);
+        this.config.setApplicationName(applicationName);
+        this.config.setConnectionTimeoutMillis(connectionTimeoutMillis);
+    }
+
+    public Client(String apiKey, Environment environment, String applicationName) {
+
+        this.config = new Config();
+        this.config.setApiKey(apiKey);
+        this.setEnvironment(environment);
+        this.config.setApplicationName(applicationName);
+    }
+
+    public Client(String apiKey, Environment environment, String applicationName, int connectionTimeoutMillis) {
+
+        this.config = new Config();
+        this.config.setApiKey(apiKey);
+        this.setEnvironment(environment);
+        this.config.setApplicationName(applicationName);
+        this.config.setConnectionTimeoutMillis(connectionTimeoutMillis);
+    }
+
```

Comment:
let's add overloads as well for ApplicationName as this should not be a mandatory field

Reply:
It is not mandatory. We can pass null value if we don't want to set applicationName. Point here is- we should not overload constructors for all possible combinations if we can pass null for some optional parameters.

Reply:
Understand, when we add applicationInfo we can remove this field so let's add only constructors without applicationName for now.